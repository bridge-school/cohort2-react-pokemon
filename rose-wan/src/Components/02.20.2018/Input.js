import React from 'react';

export const Input = ({ onInputChange, inputValue }) => {
  return <input onInputChange={onInputChange} value={inputValue} />;
};
