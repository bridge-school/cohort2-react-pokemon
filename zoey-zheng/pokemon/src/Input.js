import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updateInputValue } from './actions/index';

class Input extends Component {
  handleChange = event => {
    let input = event.target.value.toLowerCase();
    this.props.updateInputValue(input);
  };

  render() {
    return (
      <input
        autoFocus
        className="inputBox"
        ref={this.props.refProp}
        onChange={this.handleChange}
        type="text"
        placeholder="search pokemon"
      />
    );
  }
}

const mapDispatchToProps = {
  updateInputValue: updateInputValue
};

export default connect(null, mapDispatchToProps)(Input);
