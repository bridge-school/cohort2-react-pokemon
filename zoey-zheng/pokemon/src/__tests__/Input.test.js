import React from 'react';
import Input from '../Input.js';
import renderer from 'react-test-renderer';

describe('input component tests', () => {
  it('renders input component', () => {
    const input = renderer.create(<Input />);
    expect(input.toJSON()).toMatchSnapshot();
  });
});
