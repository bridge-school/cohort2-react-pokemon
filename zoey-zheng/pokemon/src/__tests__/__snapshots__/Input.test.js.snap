// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component tests renders input component 1`] = `
<input
  autoFocus={true}
  className="inputBox"
  onChange={[Function]}
  placeholder="Search Pokemon"
  required={true}
  type="text"
/>
`;
